### ISSUE TITLE
Coins become un-sendable during bidding

### ISSUE THREAD

**Issue by CryptoBeans** [particl/particl-desktop#1525]
I noticed that when I bid on an item, the balance I am allowed use on the Send page goes down by the amount I bid. For example, I have about 300 coins in my test wallet, but it says that I can only send 60 of them. I do not have any escrows in effect right now, but there are many Bidding orders open (and I assume most are lost now as they are past 48 hours and seller never responded/accepted).

Edit: I see now that my bids are actually doing a transfer as seen here: https://explorer-testnet.particl.io/tx/fcc043f52644ba0c5d0f581ac8bf0f949276ae5dab51899c2b9d173823bfdeab

How does a user get their coins back from these bidding addresses? I seem to have a lot "locked" in them and even though the associated listing I bid on is no longer valid (they've expired past the 48 hours), the coins never seemed to make their way back to being spendable within my wallet.

**Comment by CryptoBeans** [comment]
Is this possibly related to the change made for this bug? https://github.com/particl/particl-desktop/issues/1428

Either way, the coins used in bidding need to get back into a sendable state and it appears they are not doing so automatically when the associated listing for the bid expires.

**Comment by zaSmilingIdiot** [comment]
Thanks for bringing this up... as a quick note: this is not related to changes made to address #1428 . The reason for the original cause of that is the same as this, but the changes introduced there fix that specific issue, with the underlying issue not having been addressed (as yet) entirely.

Simply put, the current process of creating the escrow follows steps similar to the following (very broadly, generalized sequence):
1. Buyer bids on item: locks the relevant utxos that would be used if the bid is successful
2. Seller accepts bid: signs a transaction that includes the buyers utxos as inputs, as well as the sellers own utxos.
3. Buyer confirms escrow: signs the seller-signed transaction creating the actual transaction. This is the escrow transaction (created on the blockchain, and visible now in the transaction table).

There are a couple of concerns with this, one of which you've pointed out: the buyer locks utxos, and so while the amounts are not spent as yet, they are locked and are not available for spending. Another issue is that if the same wallet is used on a different device, then the "locked funds" on the one device are actually spendable on the 2nd device (which causes problems with bid acceptance/escrow locking).

This whole process should soon be completed being updated: the change to include the omp-lib should allow for the transaction process to start when the seller accepts a bid. That is, in a very simplified description: the seller signs a bid transaction which the buyer then signs to create the actual escrow transaction. In other words, there are no locking of funds until the bid is accepted.
See for example the tests defined here: https://github.com/particl/omp-lib/blob/master/__tests__/buyflow-multisig.test.ts

So to summarize: the locking of funds when placing a bid is due to the current implementation of creating a transaction. This very soon, however, should no longer be an issue.

Finally, on a related note: the issue of identifying spendable amounts vs locked amounts is being addressed in https://github.com/particl/particl-desktop/pull/1453 . In summary, we are adding a 'spendable' value to the Overview page, as well as adding in a 'Spendable', Locked' and 'Total' balance when hovering over the icon in Particl Desktop. This should make it easy to identify the actual amounts available for spending whenever it may be applicable (although it would be less applicable once the changes with omp-lib have been included).

**Comment by CryptoBeans** [comment]
@zaSmilingIdiot Thanks for the thorough explanation. I'm going to close this issue because it looks like you guys are fully aware of what I identified here and have a solution in progress.
